pkg/kv/kvclient/rangefeed/rangefeed.go

```diff
@@ -291,7 +304,7 @@ func (f *RangeFeed) run(ctx context.Context, frontier span.Frontier) {
 		if failed := f.runInitialScan(ctx, &restartLogEvery, &r, frontier); failed {
 			return
 		}
-	} else {
+	} else if !resumeWithFrontier {
```

Comment:
Do we need this conditional here? what breaks if we unconditionally forward to the initial ts here?

Reply:
nothing breaks, just a tiny perf optimization that may or may not matter.